internal/httpproxy/http_proxy.go

```diff
@@ -124,6 +124,9 @@ func (p *ProxyConn) authenticate() error {
 		return err
 	}
 
+	// Store reference to original metrics wrapper before TLS upgrade
+	originalMetricsConn, hasMetrics := p.Conn.(*connWithMetrics)
```

Comment:
The reason we need to store the `originalMetricsConn` is because in line 131, we upgrade the connection to TLS. 

The `tlsConnectConn.conn` has the `connWithMetrics` but we can't access unexported `tls.conn` field.

Reply:
Regardless of changing the underlying connection `p.Conn`, I think this type casting is not very nice because `ProxyConn` creates this `connWithMetrics` in the constructor. I would add an explicit field to reference the underlying TCP connection e.g.
```go
type ProxyConn struct {
    Conn net.Conn
    tcpConn *connWithMetrics
    // other fields...
}
```

Reply:
I agree that type assertion is not nice here too. If we add `tcpConn *connWithMetrics` for reference, we'd have both `Conn` and `tcpConn` referring to the same `Conn` initially ([code](https://github.com/Twingate/kubernetes-access-gateway/pull/71/files#diff-309da065b292d8850fba689660bc96fadda998d8562fd3ad8f1c6930cb30ac61R261-R262)). I think it's not the best either? But I don't have a better suggestion :s

Reply:
> I think it's not the best either?

Not ideal but I think it's the best we could do here 🙈

Reply:
Yes, that's a great idea! This `connWithMetrics` is pretty coupled with the `ProxyConn`.

Reply:
Yeah, I think this approach is nicer. One small thing to note is that we still need to implement type assertion [here](https://github.com/Twingate/kubernetes-access-gateway/pull/71/files#diff-309da065b292d8850fba689660bc96fadda998d8562fd3ad8f1c6930cb30ac61R380-R385). I suppose we cannot avoid type assertion regardless...